server/balancer.go

```diff
@@ -181,33 +182,35 @@ func (r *replicaChecker) Check(region *regionInfo) Operator {
 		return newRemovePeer(region, oldPeer)
 	}
 
-	return r.checkBetterPeer(region)
+	return r.checkBestReplacement(region)
 }
 
 // selectBestPeer returns the best peer in other stores.
 func (r *replicaChecker) selectBestPeer(region *regionInfo, filters ...Filter) (*metapb.Peer, float64) {
-	filters = append(filters, r.filters...)
+	// Add some must have filters.
+	filters = append(filters, newStateFilter(r.opt))
```

Comment:
`healthFilter` will be used after the best store has been found, or the best store can be filtered if it is temporarily unavailable.